xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/extensionHistory.vm

```diff
@@ -138,6 +138,7 @@
             <input type="hidden" name="xredirect" value="$xredirect" />
           </div>
           <input type="file" name="filepath" />
+          <input type="submit" name="submit" class="hidden" />
```

Comment:
This will make the WCAG test pass but it doesn't mean we satisfy the WCAG rule. If we want to satisfy the rule then we should do something similar to what is done on the Attachments tab:

* we generate a visible submit button from HTML
* we hide the submit button from JavaScript

The effect is that:

* when JavaScript is enabled the file is uploaded automatically without the need to click on the submit button (which is hidden)
* when JavaScript is disabled you need to click on the submit button to upload the file

For this we need to:

* have a proper label (``value``) on the submit button (even if it's hidden)
* no need for a ``name`` since we don't need to have the button name submitted
* hide the button from JavaScript when the upload form is enhanced.

Reply:
OK for going in this direction. Just a side question: isn't the name of an input mandatory for a WCAG rule?

Reply:
I'm not aware of such a rule. What I know is that the ``name`` is needed if you want to push some data to the server. For instance, if you have multiple submit buttons and you want to know on the server side which one was clicked then you need to use the ``name`` attribute so that the button label (``value``) is sent to the server. Otherwise, if you have a single submit button then you don't need the button label on the server (most of the time).